### PR TITLE
Fixes THRIFT-3855

### PR DESCRIPTION
**Pull request by asm0dey**
Replaced complex CompareAndSwapInt64 with syncOnce.

### CODE REVIEW

File: lib/go/thrift/simple_server.go [hunk 1]
```diff
@@ -183,9 +186,6 @@ func (p *TSimpleServer) processRequests(client TTransport) error {
 			log.Printf("error processing request: %s", err)
 			return err
 		}
-		if err, ok := err.(TApplicationException); ok && err.TypeId() == UNKNOWN_METHOD {
-			continue
-		}
```
Comment (Jens-G) [inline lib/go/thrift/simple_server.go]: I can't see the connection. Please explain.

Comment (Jens-G) [inline lib/go/thrift/simple_server.go]: How is that related? Please explain.

Comment (asm0dey) [inline lib/go/thrift/simple_server.go]: It's removed by gofmt. I thought that gofmt knows better tan me, so I didn't revert that change. should I?

Comment (asm0dey) [inline lib/go/thrift/simple_server.go]: @Jens-G --^

Comment (Jens-G) [inline lib/go/thrift/simple_server.go]: I would doubt that. If that is true that would probably qualify for a bug report. I would not expect a formatting tool to arbitrarily delete valid code lines.

Comment (asm0dey) [inline lib/go/thrift/simple_server.go]: @Jens-G  Reverted back that change. Thank you for review!

Comment (asm0dey) [inline lib/go/thrift/simple_server.go]: And again I dunno why one of builds has failed, but it looks like it's nothing to do with Fo code.
